src/server/api.test.js

```diff
@@ -0,0 +1,74 @@
+import request from 'supertest';
+
+import app from '.';
+import * as chartData from '../utils/apiUtils/chartData';
+
+describe('GET /api/stocks/:symbol', () => {
+  let mockRequestMaxStockData;
+  let mockRequestQuote;
+  let mockRequestOneDayStockData;
+  let mockRequestFiveDayStockData;
+
+  beforeAll(() => {
+    const mockRequestImplementation = (_symbol, callback) => (
+      callback(null, {})
+    );
+
+    mockRequestMaxStockData = jest.spyOn(
+      chartData, 'requestMaxStockData'
+    ).mockImplementation(mockRequestImplementation);
+
+    mockRequestQuote = jest.spyOn(
+      chartData, 'requestQuote'
+    ).mockImplementation(mockRequestImplementation);
+
+    mockRequestOneDayStockData = jest.spyOn(
+      chartData, 'requestOneDayStockData'
+    ).mockImplementation(mockRequestImplementation);
+
+    mockRequestFiveDayStockData = jest.spyOn(
+      chartData, 'requestFiveDayStockData'
+    ).mockImplementation(mockRequestImplementation);
+  });
+
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  let server;
+
+  beforeEach(() => {
+    server = app.listen(3000);
+  });
+
+  afterEach(done => {
+    server.close(done);
+  });
+
+  it('returns 200 for a valid stock symbol', () => {
+    const mockSymbolValid = 'MSFT';
+    return request(server).get(`/api/stocks/${ mockSymbolValid }`)
+      .then(res => {
+        expect(res.statusCode).toBe(200);
+      });
+  });
+  it('returns 404 for an invalid stock symbol', () => {
+    mockRequestQuote.mockImplementationOnce((_symbol, callback) => {
+      callback('error');
```

Comment:
Technically you only need to call `callback('error')` in 1 out of the 4 places you do to get the desired result, but this way works too I suppose